app/models/book.rb

```diff
@@ -1,3 +1,6 @@
 class Book < ApplicationRecord
   validates :genre, :author, :image, :title, :publisher, :year, presence: true
+
+  has_one :rent, dependent: :destroy
+  has_one :user, through: :rent
```

Comment:
This is okay but not necessary for now, I recommend to delete it. 
I'm not pretty sure but could bring some problems.